blackbox/docs/administration/ssl.txt

```diff
@@ -114,3 +114,272 @@ string. E.g.::
+      [Unknown]:  AT
+    Is CN=CrateDB, OU=PSQL Netty SSL, O=crate.io, L=Dornbirn, ST=Vorarlberg, C=AT correct?
+      [no]:  yes
+
+    Enter key password for <server>
+        (RETURN if same as keystore password):
+    Re-enter new password:
+
+
+Generate a Certificate Signing Request
+--------------------------------------
+
+To establish trust for this key, we need to sign it. This is done by generating
+a certificate signing request. If you have access to a certificate authority (CA),
+you can skip the next optional steps and get the signed certificate from it using
+the signing request which we will generate in the following. If you don't, then
+follow the optional steps after this step to establish your own CA.
+
+Command::
+
+    keytool -keystore keystore -certreq -alias server -keyalg rsa -file server.csr
+
+
+Output::
+
+    Enter keystore password:
+    Enter key password for <server>
+
+
+Optional: Generate a Self-Signed Certificate to act as a Certificate Authority (CA)
+-----------------------------------------------------------------------------------
+
+If you don't get your certificate signed from one of the official CAs,
+you might want to create your own CA with a self-signed certificate:
+
+Command::
+
+    openssl req -x509 -sha256 -nodes -days 36500 -newkey rsa:2048 -keyout rootCA.key -out rootCA.crt
+
+
+Output::
+
+    Generating a 2048 bit RSA private key
+    .......................................................................+++
+    .............................................................+++
+    writing new private key to 'rootCA.key'
+    -----
+    You are about to be asked to enter information that will be incorporated
+    into your certificate request.
+    What you are about to enter is what is called a Distinguished Name or a DN.
+    There are quite a few fields but you can leave some blank
+    For some fields there will be a default value,
+    If you enter '.', the field will be left blank.
+    -----
+    Country Name (2 letter code) [AU]:AT
+    State or Province Name (full name) [Some-State]:Vorarlberg
+    Locality Name (eg, city) []:Dornbirn
+    Organization Name (eg, company) [Internet Widgits Pty Ltd]:Crate.io
+    Organizational Unit Name (eg, section) []:Cryptography Department
+    Common Name (e.g. server FQDN or YOUR name) []:crate.io
+    Email Address []:info@crate.io
+
+
+Optional: Generate Signed Cert
+------------------------------
+
+Now it is time to generate a signed cert from our certificate signing request.
+
+Command::
+
+    openssl x509 -req -in server.csr -CA rootCA.crt -CAkey rootCA.key -CAcreateserial -out server.crt -sha256 -days 36500
+
+Output::
+
+    Signature ok
+    subject=/C=AT/ST=Vorarlberg/L=Dornbirn/O=crate.io/OU=PSQL Netty SSL/CN=CrateDB
+    Getting CA Private Key
+
+
+Optional: Import CA into Keystore
+---------------------------------
+
+The CA needs to be imported to the Keystore for the certificate chain to be
```

Comment:
Should this really be lower case? I believe this is a special name and is capitalized: https://en.wikipedia.org/wiki/Keystore